pkg/validations/validate.go

```diff
@@ -163,30 +192,54 @@ func (v validationKwargs) shouldValidate(value starlark.Value, thread *starlark.
 	return true, nil
 }
 
+func (v validationKwargs) populateArgs(value starlark.Value, parent starlark.Value, root starlark.Value) ([]starlark.Tuple, error) {
+	kwargs := []starlark.Tuple{}
+	if whenFunc, isFunc := v.when.(*starlark.Function); isFunc {
+		for idx := 0; idx < whenFunc.NumParams(); idx++ {
+			name, _ := whenFunc.Param(idx)
+			switch name {
+			case "v", "val", "value":
+				kwargs = append(kwargs, starlark.Tuple{starlark.String(name), value})
+			case "p", "par", "parent":
+				kwargs = append(kwargs, starlark.Tuple{starlark.String(name), parent})
+			case "r", "root", "d", "doc", "document":
+				// When validation support is extended beyond Data Values and into documents in general, differentiate
+				//   between "root" and "document":
+				//   - since data values validation occurs on a document, "root" and "document" are synonyms, right now.
+				//   - once validations can be defined in templates, then "root" will be a DocSet.
+				kwargs = append(kwargs, starlark.Tuple{starlark.String(name), root})
```

Comment:
ive been chewing on this a bit more today... care to do a quick zoom with me to talk it through?

Reply:
I would love to! I'll reach out.

Reply:
Summary:
- chief concern is that nowhere else in `ytt` is there an interaction like this: where the name of the argument of a supplied function has meaning outside of what the author wants.
- we landed on
  - one required positional parameter; it will receive the value being validated
  - a second optional positional parameter; when specified will be populated with a `struct` that has two attributes: `parent` and `root`.